keras/models.py

```diff
@@ -856,7 +856,9 @@ def load_weights(self, filepath):
 
     def fit_generator(self, generator, samples_per_epoch, nb_epoch,
                       verbose=1, show_accuracy=False, callbacks=[],
-                      validation_data=None, class_weight=None, nb_worker=1):
+                      val_generator=None,
```

Comment:
In terms of API, I think `validation_data` should be overloaded to support passing a generator. No need for a need keyword argument.